src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -1089,6 +1093,7 @@ Copyright (c) .NET Foundation. All rights reserved.
       SelfContained="$(SelfContained)"
       IsExecutable="$(_IsExecutable)"
       ReferencedProjects="@(_MSBuildProjectReferenceExistent)"
+      MSBuildVersion="$(MSBuildVersion)"
```

Comment:
Could we pass the value of `_UseAttributeForTargetFrameworkInfoPropertyNames` in here instead of the MSBuild version?  Then we wouldn't have two places where we're checking whether we're on version 17 or higher.